src/meta_memcache/serializer.py

```diff
@@ -53,3 +56,175 @@ def unserialize(self, data: Blob, encoding_id: int) -> Any:
+    _default_compression_threshold: int
+    _dict_compression_threshold: int
+    _zstd_compressors: Dict[int, zstd.ZstdCompressor]
+    _zstd_decompressors: Dict[int, zstd.ZstdDecompressor]
+    _domain_to_dict_id: Dict[str, int]
+    _default_zstd_compressor: Optional[zstd.ZstdCompressor]
+
+    def __init__(
+        self,
+        pickle_protocol: int = DEFAULT_PICKLE_PROTOCOL,
+        compression_level: int = DEFAULT_COMPRESSION_LEVEL,
+        compression_threshold: int = DEFAULT_COMPRESSION_THRESHOLD,
+        dict_compression_threshold: int = DEFAULT_DICT_COMPRESSION_THRESHOLD,
+        dictionary_mappings: Optional[List[DictionaryMapping]] = None,
+        default_dictionary: Optional[bytes] = None,
+        default_zstd: bool = True,
+    ) -> None:
+        self._pickle_protocol = pickle_protocol
+        self._compression_level = compression_level
+        self._default_compression_threshold = (
+            compression_threshold
+            if not default_dictionary
+            else dict_compression_threshold
+        )
+        self._dict_compression_threshold = dict_compression_threshold
+        self._zstd_compressors = {}
+        self._zstd_decompressors = {}
+        self._domain_to_dict_id = {}
+
+        compression_params = zstd.ZstdCompressionParameters.from_level(
+            compression_level,
+            format=zstd.FORMAT_ZSTD1_MAGICLESS,
+            write_content_size=True,
+            write_checksum=False,
+            write_dict_id=True,
+        )
+
+        if dictionary_mappings:
+            for dictionary_mapping in dictionary_mappings:
+                dict_id, zstd_dict = self._build_dict(dictionary_mapping.dictionary)
+                self._add_dict_decompressor(dict_id, zstd_dict)
+                if dictionary_mapping.active_domains:
+                    # The dictionary is active for some domains
+                    self._add_dict_compressor(dict_id, zstd_dict, compression_params)
+                    for domain in dictionary_mapping.active_domains:
+                        self._domain_to_dict_id[domain] = dict_id
+
+        if default_dictionary:
+            dict_id, zstd_dict = self._build_dict(default_dictionary)
+            self._add_dict_decompressor(dict_id, zstd_dict)
+
+            self._default_zstd_compressor = self._add_dict_compressor(
+                dict_id, zstd_dict, compression_params
+            )
+        elif default_zstd:
+            self._default_zstd_compressor = zstd.ZstdCompressor(
+                compression_params=compression_params
+            )
+        else:
+            self._default_zstd_compressor = None
+
+        self._zstd_decompressors[0] = zstd.ZstdDecompressor()
+
+    def _build_dict(self, dictionary: bytes) -> Tuple[int, zstd.ZstdCompressionDict]:
+        zstd_dict = zstd.ZstdCompressionDict(dictionary)
+        dict_id = zstd_dict.dict_id()
+        self._zstd_decompressors[dict_id] = zstd.ZstdDecompressor(dict_data=zstd_dict)
```

Comment:
Don't like to much to perform actions in instance variables through code called from outside. Maybe passing the instance variable as a dependency in the function to avoid side effects? Or return the decompressor and assign it to the instance variable

Reply:
I was moving it to `_add_dict_decompressor` but forgot to finish :D